mithril-common/benches/merkle_tree.rs

```diff
@@ -0,0 +1,75 @@
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use mithril_common::crypto_helper::{MKTree, MKTreeNode};
+
+const K: usize = 1_000;
+const M: usize = 1_000 * K;
+const TOTAL_LEAVES_BENCHES: &[usize] = &[K, 10 * K, 100 * K, M, 10 * M];
+
+fn generate_merkle_tree(total_leaves: usize) -> MKTree {
+    MKTree::new(
+        (0..total_leaves)
+            .map(|i| format!("bench-{i}").into())
+            .collect::<Vec<MKTreeNode>>()
+            .as_slice(),
+    )
+    .unwrap()
+}
+
+fn create_merkle_tree_root_benches(c: &mut Criterion) {
+    let mut group = c.benchmark_group("create_merkle_tree");
+    for total_leaves in TOTAL_LEAVES_BENCHES.iter() {
+        group.bench_with_input(
+            BenchmarkId::from_parameter(total_leaves),
+            total_leaves,
+            |b, &total_leaves| {
+                b.iter(|| {
+                    let mk_tree = generate_merkle_tree(total_leaves);
+                    mk_tree.compute_root().unwrap();
+                });
+            },
+        );
+    }
+    group.finish();
+}
+
+fn create_merkle_tree_proof_benches(c: &mut Criterion) {
+    let mut group = c.benchmark_group("create_merkle_tree_proof");
+    for total_leaves in TOTAL_LEAVES_BENCHES.iter() {
+        group.bench_with_input(
+            BenchmarkId::from_parameter(total_leaves),
+            total_leaves,
+            |b, &_total_leaves| {
+                b.iter(|| {
+                    let mk_tree = generate_merkle_tree(*total_leaves);
+                    let leaves_to_prove = mk_tree.leaves();
```

Comment:
It's may be possible to generalize `create_merkle_tree_root_benches`and `create_merkle_tree_proof_benches` passing a lambda. It's may be inappropriate because of type returning by `compute`.